spot.go

```diff
@@ -58,7 +60,7 @@ func (l *SpotListener) Start(ctx context.Context, notices chan<- TerminationNoti
 				log.Error("Empty response from metadata")
 				continue
 			}
-			t, err := time.Parse("2006-01-02T15:04:05Z", out)
+			t, err := time.Parse(time.RFC3339, out)
```

Comment:
Are you sure this matches RFC3339? I recall pulling my hair out because it was a subtle variant.

Reply:
I haven't tested this with an actual spot termination, but according to the docs it should work:
- RFC3339 is defined as `2006-01-02T15:04:05Z07:00` here: 
https://golang.org/pkg/time/#pkg-constants
- The EC2 metadata documentation also has an [example](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html) of a spot termination timestamp:
 `2015-01-05T18:02:00Z`

The only gotcha that I'm aware of is that Go is strictly compliant with RFC3339, so it either accepts `Z` for Zulu time _or_ `+00:00`, but not both together (which I've seen sometimes, but not on AWS). That being said, after some quick tests the only difference `2006-01-02T15:04:05Z` and `time.RFC3339` is that `time.RFC3339` supports timestamps with offsets and not just Z for Zulu time.

tl;dr I think this should be fine, and we can always roll back if it does not work 👍